### PR TITLE
#162700971 ft(schema): user model-schema

### PR DESCRIPTION
**Pull request by Abudu-Samuel**
### What does this PR do?
Sets up collection schema for users model

### Description of Task to be completed?

- [x] add schema for model user
- [x] add bcyrpt to hash and compare passwords

### How should this be manually tested?

Nill

Any background context you want to provide?

Nil

### What are the relevant pivotal tracker stories?

story type: chore
story Id: #162700971
story title: Setup collection schema

### Screenshots (if appropriate)
N/A

### CODE REVIEW

File: server/models/User.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  updatedAt: {
+    type: Date
+  }
+});
+
+UserSchema.methods.generateHash = password => hashSync(password, genSaltSync(8), null);
+
+UserSchema.methods.isValidPassword = password => compareSync(password, this.password);
+
+const UserModel = mongoose.model('User', UserSchema);
```
Comment (chykehyman) [inline server/models/User.js]: Try doing `Export Default` directly on this line

**Review by chykehyman — CHANGES_REQUESTED** [review]

File: server/models/User.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
+import { Schema, model } from 'mongoose';
+import { hashSync, genSaltSync, compareSync } from 'bcrypt';
+
+const UserSchema = new Schema({
```
Comment (chykehyman) [inline server/models/User.js:4]: Try importing `Schema` from mongoose and using it directly here

**Review by chykehyman — CHANGES_REQUESTED** [review]

File: server/models/User.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  updatedAt: {
+    type: Date
+  }
+});
+
+UserSchema.methods.generateHash = password => hashSync(password, genSaltSync(8), null);
+
+UserSchema.methods.isValidPassword = password => compareSync(password, this.password);
+
+export const UserModel = model('User', UserSchema);
```
Comment (chykehyman) [inline server/models/User.js]: Can we make the collection name start with a lowercase? i.e. instead of `User`, write `user`

**Review by chykehyman — CHANGES_REQUESTED** [review]

File: server/models/User.js (hunk 1 above)
Comment (Abudu-Samuel) [reply server/models/User.js]: Thanks for pointing that out, will go ahead to make necessary adjustment

File: server/models/User.js (hunk 2 above)
Comment (Abudu-Samuel) [reply server/models/User.js:4]: Thanks, fully noted!

File: server/models/User.js (hunk 3 above)
Comment (Abudu-Samuel) [reply server/models/User.js]: Hi, thanks for seeing that, I thought as much, i just wasn't sure

**Review by chykehyman — APPROVED** [review]

**Pull request merged**